src/compiler/checker.ts

```diff
@@ -6850,13 +6870,7 @@ namespace ts {
                 const propType = getTypeOfSymbol(p);
                 const widenedType = getWidenedType(propType);
                 if (propType !== widenedType) {
-                    const symbol = <TransientSymbol>createSymbol(p.flags | SymbolFlags.Transient, p.name);

```

Comment:
I remember that when I looked at this bug, I tried to unify the behavior between `getWidenedTypeOfObjectLiteral` and `getRegularTypeOfObjectLiteral` because you do the same thing - namely, you recursively grab the widened/regular type of each  property. In the end, things seemed pretty clean and I was able to share a lot of the same logic.

I don't remember what specifically I was running into that my fix didn't address. I'd try to play around with how deep you nest these tests.